### PR TITLE
feat!: make submitPFB broadcast mode configurable

### LINKED ISSUE #1885: blob: SubmitPayForBlob should allow configurable broadcast mode
Or use the `sync` instead of `block`. 

As per @evan-forbes, `block` is going to be deprecated soonish, so we should stop using it. Additionally, it causes problems for the Dymension team

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

closes #1885

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by rootulp — APPROVED** [review]

**Review by Wondertan — COMMENTED** [review]
[non-blocking]
Can this be an option with a sane default?

**Review by Wondertan — APPROVED** [review]

**Comment by evan-forbes** [comment]
> Can this be an option with a sane default?

we could use options or something similar, but we are already accepting TxbuilderOptions so I didn't want to add two sets of options here. This function does have too many args tho, so its due for a refactor

**Comment by Wondertan** [comment]
Ok, I was wondering if you can extend existing options.

File: app/test/integration_test.go [hunk 1]
```diff
@@ -262,7 +263,14 @@ func (s *IntegrationTestSuite) TestSubmitPayForBlob() {
 			require.NoError(t, s.cctx.WaitForBlocks(3))
 
 			signer := blobtypes.NewKeyringSigner(s.cctx.Keyring, s.accounts[141], s.cctx.ChainID)
-			res, err := blob.SubmitPayForBlob(context.TODO(), signer, s.cctx.GRPCClient, []*blobtypes.Blob{tc.blob, tc.blob}, tc.opts...)
+			res, err := blob.SubmitPayForBlob(
+				context.TODO(),
+				signer,
+				s.cctx.GRPCClient,
+				sdktx.BroadcastMode_BROADCAST_MODE_BLOCK,
```
Comment (SweeXordious) [inline app/test/integration_test.go:270]: Shouldn't this be broadcast mode sync? As per the issue?

Comment (evan-forbes) [reply app/test/integration_test.go:270]: it was always block before, so I left it as is for now

**Review by cmwaters — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]

**Comment by evan-forbes** [comment]
> Ok, I was wondering if you can extend existing options.

we could be it felt a bit too hacky and temporary. actually reducing the complexity and number of arguments I think requires a full refactor. Like, we'd have to wrap the txbuilder options in this more abstract option type, and apply the option two different times, once to see if there are tx builder options present when building the tx, then another time before we submit the tx. The alternative being to just add an extra argument, which felt like the least worst option for the time being

**Review by staheri14 — APPROVED** [review]
LGTM!

**Pull request merged**